File: examples/perl/Employee.pl
#!/usr/bin/perl

package Employee; {
  sub new {
    my $class = shift;
    my $self = {
      _name     => shift,
      _salary   => shift,
    };
    bless $self, $class;
    return $self;
  }
    
  sub displayEmployee {
    my( $self ) = @_;
    printf "Name: %s, Salary: \$%0.2f\n", $self->{_name},$self->{_salary};
  }
}

sub main {
  my $emp1 = new Employee("Bob", 75000);
  my $emp2 = new Employee("Joe", 50000);
  $emp1->displayEmployee();
  $emp2->displayEmployee();
  system($?>>8);
}

main();

File: examples/ruby/Employee.rb
#!/usr/bin/ruby

class Employee
  attr_accessor :name, :salary
  
  def initialize(name, salary)
    @name = name
    @salary = salary
  end

  def to_s
    return "Name: %s, Salary: $%.2f" % [@name, @salary]
  end
  
  def displayEmployee
    puts self
  end
end

def main
  emp1 = Employee.new("Bob", 75000)
  emp2 = Employee.new("Joe", 50000)
  emp1.displayEmployee()
  emp2.displayEmployee()
  exit(0)
end
 
if __FILE__ == $PROGRAM_NAME
  main()
end
